### PR TITLE
feat: add Dovetail connector

### PR DESCRIPTION
**Pull request by sanjay-a01**
## Checklist
- [x] Ran Linter
- [x] Create PR from non-main branch

## Notes
It supports pagination

### GET
URL: <http://localhost:4444/api/v1/notes>
image

### GET notes by id
URL: <http://localhost:4444/api/v1/notes/1IxNQrhqWFPzyZ45jFXkM1> 
image

### POST
URL: <http://localhost:4444/api/v1/notes> 
image

### PATCH
URL: <http://localhost:4444/api/v1/notes/5mntv5PjL3CLHddiBSnNNx>
image

### DELETE
URL:<http://localhost:4444/api/v1/notes/5mntv5PjL3CLHddiBSnNNx>
image

### PAGINATION
Requesting a paginated list of notes using page[limit] in the param
URL:<http://localhost:4444/api/v1/notes?page[limit]=5>
image

Requesting a paginated list of notes using page[start_cursor] in the param will give the results of the next page
URL:<http://localhost:4444/api/v1/notes?page[start_cursor]=WyJjcmVhdGVkX2F0X2Rlc2MiLFsiMjAyNC0wOS0xNlQxNDowNjoyMy44NDQzNzUrMDA6MDAiLCJjYmM1YTk3Mi1lMDdlLTRhMzQtYWEyNS1mNzk2MjRjMzAyZGQiXV0>
image

### CODE REVIEW

File: providers/dovetail.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package providers
+
+const Dovetail Provider = "dovetail"
+
+func init() {
+	SetInfo(Dovetail, ProviderInfo{
+		DisplayName: "Dovetail",
+		AuthType:    ApiKey,
+		BaseURL:     "https://dovetail.com",
```
Comment (RajatPawar) [inline providers/dovetail.go:9]: Any reason this isn't `https://dovetail.com/api`?

**Review by RajatPawar — APPROVED** [review]
Approving, but please resolve the comment before merging

File: providers/dovetail.go (hunk 1 above)
Comment (sanjay-a01) [reply providers/dovetail.go:9]: No specific reason. I'll check with the docs and update it.

**Comment by sanjay-a01** [comment]
I have updated the  base URL in Dovetail

**Pull request merged**